src/core/agent/records/connectionRecord.ts

```diff
@@ -16,6 +16,7 @@ class ConnectionRecord extends BaseRecord {
   oobi!: string;
   groupId?: string;
   pending!: boolean;
+  pendingDeletion?: boolean;
```

Comment:
Why is this optional? We only set it as optional if it really is optional. It's false by default.

If you set it as optional, TS will think it can be undefined in other places when retrieving the record.